### ISSUE TITLE
PDF Tooling

### ISSUE THREAD

**Issue by russbiggs** [hotosm/projects#28]
Currently focusing on [pandoc](https://pandoc.org).  It seems well maintained compared to other solutions. I'm currently exploring the different PDF rendering options.  A plus is that it can also do markdown --> latex (this is actually the invisibile intermediate step for PDF as well)

It works as a CLI tool but given that the PDF processing will be offloaded to CI it should be easy to install on the circleci machine.

**Comment by russbiggs** [comment]
@smit1678

**Comment by smit1678** [comment]
Great thanks @russbiggs. Yeah I agree that the best and most widely used is Pandoc. I think the issue with Pandoc is always Latex installation but I imagine that can be solved with some dockerized setups, and it looks like its there: https://github.com/pandoc/dockerfiles

Here's a few links from some initial research I started: 

https://wkhtmltopdf.org/
https://blog.scottlowe.org/2018/09/27/a-markdown-to-pdf-workflow-on-linux/
http://mixu.net/markdown-styles/
https://atom.io/packages/markdown-themeable-pdf

**Comment by russbiggs** [comment]
I agree the install can be messy.  Since this is going to be put onto a circleci process we can pay that installation tax up front with the deployment set up.  One option is to the maintain the set up in the circleci config or have a dockerhub image prebuilt for use by Circleci.  This way the end user doesn’t need to mess with an my installation issues on their machine

**Comment by russbiggs** [comment]
pandoc is working well.  Allows a good amount of styling such as using the Archivo font and page sizing.  Using the [dalibo/pandocker](https://github.com/dalibo/pandocker) docker image on the circleci side makes the deployment pretty straightforward.

**Comment by russbiggs** [comment]
Sample PDF:
[1-1-mapping-project-workflow.pdf](https://github.com/hotosm/projects/files/3100746/1-1-mapping-project-workflow.pdf)

@smit1678 @jessbeutler

**Comment by jessbeutler** [comment]
Font styling looks nice. Is it possible to have headers/footers? (i.e. logo, source url, etc.)

**Comment by russbiggs** [comment]
I haven't tinkered with it yet but yes you can change the default rendering template. It would be best to bundle a generic template with header/footer with the core project so that the build process isnt too tied to a single style.  If you give me a sample of a layout I can try to get an output example.

**Comment by smit1678** [comment]
Now that we have a v1 of the PDF tooling, let's revisit where this can go. Couple improvements on my end for discussion: 

* PDF size. Current content sites are quite large: 50+ mb. 
* Table of Contents (is this possible?)
* header or footer content
* rendering tables

@jessbeutler @adisadit @russbiggs @cgiovando What else do we want to discuss here for evolving the PDF process?

**Comment by russbiggs** [comment]
@smit1678 - at one point we discussed cover pages for the PDFs. I'm not totally sure how it could fit into the workflow but perhaps worth exploring if still interested
